### ISSUE TITLE
Plugin does not provide configuration file in expected location

### ISSUE THREAD

**Issue by ryu-ketsueki** [esjeon/kwin-forceblur#23]
The title above is an error given whenever I try to open the configuration dialog. It happens with Force Blur only. I was going to use it to blur Firefox as I got a CSS theme that has a transparent background

**Comment by arun54321** [comment]
Did you find a way to fix it?
![image](https://user-images.githubusercontent.com/42943340/153532214-c263f74c-c39e-471c-a616-14bceb9e8538.png)

**Comment by lucasrafael98** [comment]
Just for visibility, the config for Force Blur is in `~/.config/kwinrc`. There should be something like
```
[Script-forceblur]
blurContent=true
patterns=yakuake\nurxvt\nkeepassxc\n
```
Assuming you need to add another program, add to the `patterns`, separating everything with a \n.

**Comment by EysseW** [comment]
I have got the same problem. The piece of text you mentioned doesn't show up in the kwinrc.

**Comment by lucasrafael98** [comment]
The script was already working on my machine before an update returned this error. I'd try adding those lines to the kwinrc and running `kwin_x11 --replace` to check that the script is loaded. The logs should have:

```
qml: FORCE-BLUR: starting the script
qml: ["yakuake","urxvt","keepassxc"]
```

Otherwise, not really aware of any fix for this. The last commit in this project was nearly two years ago, so I'm not sure if this will be fixed.

**Comment by EysseW** [comment]
I'll just use a WM :)

**Comment by ryu-ketsueki** [comment]
I reported a bug directly to KDE some time ago and I got an answer. It would seem that with Kwin's migration to jsons for metadata, the .desktops got not only deprecated but also unsupported. Force Blur must move from using .desktop for metadata to json.

https://develop.kde.org/docs/extend/plasma/widget/setup/#metadatajson

**Comment by esjeon** [comment]
It sounds like the source of the error message is here:

https://invent.kde.org/plasma/kwin/-/blob/e4ac215fcb4b364a53575638f5212966e370f191/scripting/genericscriptedconfig.cpp#L81

So KCM checks for two file: `kconfigXTFile` (= `main.xml`) and `uiPath` = `config.ui`. Manually punching in the variables, I could verify that the files indeed exist. It's possible that something set those variables wrong in the path. I think I'll have to figure out how to debug KCM...

**Comment by esjeon** [comment]
Alright, I could run KWin script module independently using `kcmshell5 kwinscripts`, and got a very interesting warning message:
```
kf.kcmutils: "krohnkite.desktop" Querying the KCMs associated to a plugin using the X-KDE-ParentComponents is deprecated. Instead define the X-KDE-ConfigModule with the namespace and plugin filen name.
```

So, yeah, this script has `X-KDE-ParentComponents=forceblur` in the `metadata.desktop`, and KWin picked up appropriate config module for the plugin.
Now, this got deprecated by KWin ([Related MR](https://invent.kde.org/plasma/kwin/-/merge_requests/306?commit_id=37ba0404c8246cf7ba85b8ada97c8ffe2eef93a5)), and now plugins (effects, scripts) must designate its config module directly. Yeah, sounds better.

Anyway, I'll upload the fix soon.

**Comment by pc00per** [comment]
0.6 fixed configuration panel, but blur doesn't work.

**Comment by esjeon** [comment]
Yup you're right... The fix was fine, but an error was introduced while clearing up the metadata.desktop... Releasing *again*.

**Comment by leon-erd** [comment]
Unfortunately I still see "Plugin does not provide configuration file in expected location" even after some reinstalls of the script via the github code or directly via "Kwin Scripts" -> "Get New Scripts". 
I also executed the two commands to enable script configuration.
Am I missing something?
My setup is:
Manjaro: Qonos 21.2.3
KDE Plasma: 5.23.5

**Comment by EysseW** [comment]
I switched to i3 now but I if you want me to, I could test if the plugin will work.

**Comment by leon-erd** [comment]
Any help is appreciated! :)

**Comment by esjeon** [comment]
@leon-erd I uploaded two different versions, because KWin 5.24.0 broke the compatibility (though it seems like they fixed it back in 5.24.1). If you're using 5.23, then please download v0.5. The other one (v0.6.1) is currently broken for KWin<5.24.0.

**Comment by leon-erd** [comment]
Thanks a lot! It does work now with the v0.5! :)
I actually tried that version before but then the configuration didn't show up even after executing the two commands from the readme so I thought that this version was broken and didn't test it further.
I figured out now that you have to reboot you PC after executing the two commands in order to get the configuration panel working. Maybe you can add this to the readme :)
